app/models/shipit/stack.rb

```diff
@@ -144,6 +144,8 @@ def merge_status
         'locked'
       else
         significant_statuses = undeployed_commits.map(&:significant_status)
+        significant_statuses << last_deployed_commit.significant_status unless last_deployed_commit.blank?
```

Comment:
That confused me as well at first. But since `<<` adds the element to the end of the array, we're fine.